Assets/_BForBoss/_Entities/Scripts/PlayerLifeCycleBehaviour.cs

```diff
@@ -29,7 +32,11 @@ protected override void LifeCycleOnHeal()
 
         protected override void LifeCycleFinished()
         {
-            _onDeathCallBack.Invoke();
+            _onDeathCallback?.Invoke();
+            VisualEffectsManager.Instance.Distort(HUDVisualEffect.Death).OnComplete(() =>
+            {
+                _onEndGameCallback.Invoke();
+            });
```

Comment:
Wait so are we saying, that when the player dies, it resets but then it also calls EndGame?

Reply:
I updated it so that
1 - Death - Controls get's disabled, VFX Start
2 - End Game - that's when the end game menu shows up. 

Nothing get's reset until player presses reset